src/download.rs

```diff
@@ -65,11 +67,15 @@ where
 {
     let client_url = url.clone();
 
-    #[rustfmt::skip]
-    let mut res = client.get(url)
-        .send()
-        .await
-        .context(format!("client get and send({:?}) failed", client_url.as_str()))?;
+    let mut res = crate::retry_loop(
+        || {
+            let res = client.get(url.clone()).send();
+            Ok(res)
+        },
+        MAX_DOWNLOAD_RETRY,
+    )
+    .map_err(|err: anyhow::Error| anyhow!("client get & send failed with err {:?}", err))?
+    .await?;
```

Comment:
I think the .await should go into the loop because that's where the result gets computed.

Reply:
Yeah, it does not work in the current form:
```
target/debug/download_sysext -p ~/kinvolk/scripts/sdk_container/src/third_party/coreos-overlay/coreos-base/coreos-au-key/files/official-v2.pub.pem -m oem-azure.gz -o /var/tmp/outdir/ -u http://localhost:8000/oem-azure.gz -v
```
fails without retry when starting either nothing (connection refused) or something like `nc -l -p 8000` (doing `ctrl-c`  to trigger `connection closed before message completed`) or `python3 -m http.server` in an empty folder (to get a 404 which with the current logic is also covered by the retry - not really needed but nice for testing, so I'm not against having it the current way).

Reply:
Done with re-implementation using `tryhard` crate to get HTTP client get/send logic included in a retry function.